lib/sshutils/scp/scp_test.go

```diff
@@ -197,40 +201,93 @@ func TestReceive(t *testing.T) {
 				),
 			),
 		},
+		{
+			desc:       "regular file into different filename (rename)",
+			config:     newTargetConfig("remote_file", Flags{}),
+			args:       args("-v", "-f"),
+			source:     "file",
+			expectedFS: newTestFS(logger, newFile("remote_file", "file contents")),
+			sourceFS:   newTestFS(logger, newFile("file", "file contents")),
+		},
+		{
+			desc:       "regular file into different filename in a directory (rename)",
+			config:     newTargetConfigWithFS("dir/remote_file", Flags{}, newTestFS(logger, newDir("dir"))),
+			args:       args("-v", "-f"),
+			source:     "file",
+			expectedFS: newTestFS(logger, newDir("dir", newFile("dir/remote_file", "file contents"))),
+			sourceFS:   newTestFS(logger, newFile("file", "file contents")),
+		},
+		{
+			desc:       "directory into different directory name (rename)",
+			config:     newTargetConfig("remote_dir", Flags{Recursive: true}),
+			args:       args("-v", "-f", "-r"),
+			source:     "dir",
+			expectedFS: newTestFS(logger, newDir("remote_dir", newFile("remote_dir/file", "file contents"))),
+			sourceFS:   newTestFS(logger, newDir("dir", newFile("dir/file", "file contents"))),
+		},
+		{
+			desc:       "directory into different directory name in subdirectory (rename)",
+			config:     newTargetConfigWithFS("dir/remote_dir", Flags{Recursive: true}, newTestFS(logger, newDir("dir"))),
+			args:       args("-v", "-f", "-r"),
+			source:     "dir",
+			expectedFS: newTestFS(logger, newDir("dir/remote_dir", newFile("dir/remote_dir/file", "file contents"))),
+			sourceFS:   newTestFS(logger, newDir("dir", newFile("dir/file", "file contents"))),
+		},
 	}
 	for _, tt := range testCases {
 		tt := tt
 		t.Run(tt.desc, func(t *testing.T) {
+			logger := logger.WithField("test", tt.desc)
 			t.Parallel()
-			cmd, err := CreateCommand(tt.config)
-			require.NoError(t, err)
 
 			sourceDir := t.TempDir()
-			source := filepath.Join(sourceDir, tt.config.Flags.Target[0])
+			source := filepath.Join(sourceDir, tt.source)
 			args := append(tt.args, source)
 
-			// Source is missing, expect an error.
-			err = runSCP(cmd, args...)
```

Comment:
I factored this into a dedicated test - no need to test this for each local testcase.